artifactory/commands/python/poetry.go

```diff
@@ -95,15 +98,190 @@ func (pc *PoetryCommand) install(buildConfiguration *buildUtils.BuildConfigurati
+			log.Warn("Failed to collect published artifacts: " + err.Error())
+		}
+	}
+
+	return nil
+}
+
+// saveFlexPackBuildInfo saves FlexPack build info for jfrog-cli rt bp compatibility
+func (pc *PoetryCommand) saveFlexPackBuildInfo(buildInfo *entities.BuildInfo) error {
+	// Create build-info service
+	service := build.NewBuildInfoService()
+
+	// Create or get build
+	bld, err := service.GetOrCreateBuildWithProject(buildInfo.Name, buildInfo.Number, "")
+	if err != nil {
+		return fmt.Errorf("failed to create build: %w", err)
+	}
+
+	// Save the complete build info (this will be loaded by rt bp)
+	err = bld.SaveBuildInfo(buildInfo)
+	if err != nil {
+		return fmt.Errorf("failed to save build info: %w", err)
+	}
+
+	return nil
+}
+
+// collectPublishedArtifacts collects information about artifacts that were published
+func (pc *PoetryCommand) collectPublishedArtifacts(buildConfiguration *buildUtils.BuildConfiguration, pythonBuildInfo *build.Build, workingDir string) error {
+	// Look for built artifacts in dist/ directory
+	distDir := filepath.Join(workingDir, "dist")
+	if _, err := os.Stat(distDir); os.IsNotExist(err) {
+		log.Debug("No dist directory found, skipping artifact collection")
+		return nil
+	}
+
+	// Get module name
+	moduleName := buildConfiguration.GetModule()
+	if moduleName == "" {
+		// Try to determine module name from pyproject.toml
+		if name, err := pc.getProjectNameFromPyproject(workingDir); err == nil {
+			moduleName = name
+		} else {
+			moduleName = "poetry-module"
+		}
+	}
+
+	// Find artifacts in dist directory
+	artifacts, err := pc.findDistArtifacts(distDir)
+	if err != nil {
+		return err
+	}
+
+	if len(artifacts) > 0 {
+		log.Info(fmt.Sprintf("Found %d artifacts to add to build info", len(artifacts)))
+		// Add artifacts to build info
+		err = pythonBuildInfo.AddArtifacts(moduleName, "pypi", artifacts...)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// getProjectNameFromPyproject extracts project name from pyproject.toml
+func (pc *PoetryCommand) getProjectNameFromPyproject(workingDir string) (string, error) {
+	pyprojectPath := filepath.Join(workingDir, "pyproject.toml")
+	viper.SetConfigType("toml")
+	viper.SetConfigFile(pyprojectPath)
+	if err := viper.ReadInConfig(); err != nil {
+		return "", err
+	}
+
+	name := viper.GetString("tool.poetry.name")
+	if name == "" {
+		return "", fmt.Errorf("no project name found in pyproject.toml")
+	}
+
+	return name, nil
+}
+
+// findDistArtifacts finds and creates artifact entries for files in dist directory
+func (pc *PoetryCommand) findDistArtifacts(distDir string) ([]entities.Artifact, error) {
+	var artifacts []entities.Artifact
+
+	entries, err := os.ReadDir(distDir)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, entry := range entries {
+		if entry.IsDir() {
+			continue
+		}
+
+		filename := entry.Name()
+		// Only include wheel and tar.gz files
+		if !strings.HasSuffix(filename, ".whl") && !strings.HasSuffix(filename, ".tar.gz") {
```

Comment:
just to improve readability

```suggestion
		if !(strings.HasSuffix(filename, ".whl") || !strings.HasSuffix(filename, ".tar.gz")) {
```

Reply:
you can also reuse the function `getArtifactType` here

Reply:
yeah, great point, I will update this to use `getArtifactType`.